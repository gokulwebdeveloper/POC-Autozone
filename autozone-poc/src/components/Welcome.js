import React, { Component,Fragment } from "react";
import ReactDOM from "react-dom";


const Welcome = (props) => {
  return (
    <Fragment>
      <h1>
        Welcome To {props.name}
      </h1>     
    </Fragment>
  );
};


export default Welcome;