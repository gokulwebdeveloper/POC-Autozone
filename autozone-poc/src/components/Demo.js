import React, { Component,Fragment } from "react";
import ReactDOM from "react-dom";
import Welcome from './Welcome';

const Demo = () => {
  return (
    <Fragment>
      <Welcome name="Autozone" />
    </Fragment>
  );
};


export default Demo;