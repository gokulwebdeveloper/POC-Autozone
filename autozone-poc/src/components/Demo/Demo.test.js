import React from "react";
import { unmountComponentAtNode } from "react-dom";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Demo from "./Demo.js";
import Welcome from "../Welcome";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

describe("renders component", () => {

  it("renders without crashing", () => {
    shallow(<Demo />);
  });

  it("renders child component with props value", () => {
    const wrapper = shallow(<Demo />);
    wrapper.name("Autozone");
    wrapper.update();
    var welcomeComponent = wrapper.find(Welcome);
    expect(welcomeComponent.props().name).toEqual("Autozone");
  });

  it("renders correctly with title", () => {
    const tree = renderer.create(<Demo />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});
