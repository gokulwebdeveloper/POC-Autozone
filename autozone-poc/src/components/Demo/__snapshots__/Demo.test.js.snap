// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component renders correctly with title 1`] = `
<h1>
  Welcome To 
  Autozone
</h1>
`;
